test/pylib/scylla_cluster.py

```diff
@@ -190,7 +190,7 @@ class ScyllaServer:
     """Starts and handles a single Scylla server, managing logs, checking if responsive,
        and cleanup when finished."""
     # pylint: disable=too-many-instance-attributes
-    START_TIMEOUT = 300     # seconds
+    START_TIMEOUT = 1000     # seconds
```

Comment:
If we decide to go this way, let's increase the timeout from 5 to 10 minutes for now (600 seconds), >15 minutes is overkill IMO.

Increasing this timeout here is not enough though, we also need to increase the request timeout for the request that the test makes to the framework to add the server. The following should work:
```diff
diff --git a/test/pylib/manager_client.py b/test/pylib/manager_client.py
index eabd95828..5315d4431 100644
--- a/test/pylib/manager_client.py
+++ b/test/pylib/manager_client.py
@@ -15,7 +15,7 @@ import logging
 from test.pylib.rest_client import UnixRESTClient, ScyllaRESTAPIClient
 from test.pylib.util import wait_for
 from test.pylib.internal_types import ServerNum, IPAddress, HostID, ServerInfo
-from test.pylib.scylla_cluster import ReplaceConfig
+from test.pylib.scylla_cluster import ReplaceConfig, ScyllaServer
 from cassandra.cluster import Session as CassandraSession  # type: ignore # pylint: disable=no-name-in-module
 from cassandra.cluster import Cluster as CassandraCluster  # type: ignore # pylint: disable=no-name-in-module
 import aiohttp
@@ -156,7 +156,8 @@ class ManagerClient():
                 data['replace_cfg'] = replace_cfg._asdict()
             if cmdline:
                 data['cmdline'] = cmdline
-            server_info = await self.client.put_json("/cluster/addserver", data, response_type="json")
+            server_info = await self.client.put_json("/cluster/addserver", data, response_type="json",
+                                                     timeout=ScyllaServer.START_TIMEOUT)
         except Exception as exc:
             raise Exception("Failed to add server") from exc
         try:
diff --git a/test/pylib/rest_client.py b/test/pylib/rest_client.py
index c2ea2a513..50b1cb74f 100644
--- a/test/pylib/rest_client.py
+++ b/test/pylib/rest_client.py
@@ -100,9 +100,9 @@ class RESTClient(metaclass=ABCMeta):
 
     async def put_json(self, resource_uri: str, data: Mapping, host: Optional[str] = None,
                        port: Optional[int] = None, params: Optional[dict[str, str]] = None,
-                       response_type: Optional[str] = None) -> Any:
+                       response_type: Optional[str] = None, timeout: Optional[float] = None) -> Any:
         ret = await self._fetch("PUT", resource_uri, response_type = response_type, host = host,
-                                port = port, params = params, json = data)
+                                port = port, params = params, json = data, timeout = timeout)
         return ret
 
     async def delete(self, resource_uri: str, host: Optional[str] = None,
diff --git a/test/pylib/scylla_cluster.py b/test/pylib/scylla_cluster.py
index aaf3e6e06..a3db29b9e 100644
--- a/test/pylib/scylla_cluster.py
+++ b/test/pylib/scylla_cluster.py
@@ -190,7 +190,7 @@ class ScyllaServer:
     """Starts and handles a single Scylla server, managing logs, checking if responsive,
        and cleanup when finished."""
     # pylint: disable=too-many-instance-attributes
-    START_TIMEOUT = 300     # seconds
+    START_TIMEOUT = 600     # seconds
     start_time: float
     sleep_interval: float
     log_file: BufferedWriter
```

(please check if it works, I haven't tested)

Reply:
I included more changes above for test and pushed a new branch to trigger a CI.

Reply:
Bootstrapping should take way less than 5 minutes, even in debug mode. Otherwise we simply can't have advanced tests for topology at the bottom of our test pyramid, or we need to disable the debug mode for them.
This in turn means we will be catching bugs late in the release process, not early, as it happened now.

Debug mode is also good at indicating we have a quadratic behavior lurking somewhere. We don't know the root cause of the slow down, it could be exactly it. 

First we need to understand what is taking so long, then make a decision. It is also a question to our test pyramid that we discover it so late in the game. Shouldn't we have performance tests for RBNO? If yes, shouldn't they have caught this? 

If we don't care about debug mode performance, let's not run certain suites in debug mode at all.

Reply:
Nothing guarantees a node bootstrap in less than 5 minutes if the host runs 1000 scylla processes in the CI machine. Both Kamil and I saw the bootstrap takes less than ~30s when running the test locally.

Reply:
> Bootstrapping should take way less than 5 minutes, even in debug mode. Otherwise we simply can't have advanced tests for topology at the bottom of our test pyramid, or we need to disable the debug mode for them.

I think this is a little vague, bootstrap should take way less than 5 minutes under which conditions? Compared to what ? We should be carefull not to make such assumptions on our verry uneven packager machines, which besides of being verry different from each other, operate under different loads at different times.

> This in turn means we will be catching bugs late in the release process, not early, as it happened now.

We can always have different configurations that run in different circumstances. For example in dtest, which this testsuite becomes very much like, there are tags for "dtest-heavy" and "dtest-long". 
We should eventually not leave any bug behing (ideally), but not all bugs have the same severty.